### ISSUE TITLE
FMVs are cropped with new 'DynamicResolution' setting

### ISSUE THREAD

**Issue by Psycho-A** [elishacloud/Silent-Hill-2-Enhancements#402]
I've noticed that new 'DynamicResolution' setting makes to ignore FMVWidescreenMode and FMVWidescreenEnhancementPackCompatibility values I define, and now videos are always displayed in uniform mode. The problem of this mode is that videos are cropped in left and right side in resoloutions narrower than 16:9 (4:3, 5:4), cropped in all sides in 16:9, and cropped in top/botton if wider than 16:9 (f.e. 21:9). That is, I can't disable cropping at all! Cropping by left/right is less noticeable with FMV Enhancement Pack, because the black pillar boxes are integrated right into the videos (why?) and cropped now - but it's very well noticeable on original game FMVs or some custom ones that have no pillarboxes integrated (it crops about 11% of content). Also, native 4:3 videos (like credits.bik) are now squashed to 16:9 in all resolutions, and cropped to ~33% horizontally in 4:3 resolutions that's already really bad. Please give the ability to watch videos just in their original pixel aspect ratio (f.e. native 640x384 FMVs displays in 1.666 ratio (16:10), and 640x480 are in 1.333 (4:3)). I don't care about pillar or letter boxes -  I'd just like to see full-framed videos as they were designed...

p.s. I've noticed that main game logo is cropped or enlarged too much in 1280x1024 too...
![Screenshot_2021-03-24_025232](https://user-images.githubusercontent.com/1223276/112233774-14d77600-8c4c-11eb-8ab8-8e5e77099ddf.png)

**Comment by Polymega** [comment]
We have been discussing ways to completely revamp the FMV display feature for the project, so all of this shouldn't be a concern in the near future.

If all goes to plan, with this revamp, FMV display modes will behave similar to [`FullscreenImages`](http://enhanced.townofsilenthill.com/SH2/config.htm#FullscreenImages) and it will be consolidated into one feature (so instead of `FMVWidescreenMode` and `FMVWidescreenEnhancementPackCompatibility`, it will be just one feature like `FMVDisplayMode`). 

When this revamp happens, you will also need to download an update to the FMV Enhancement Pack.

> Cropping by left/right is less noticeable with FMV Enhancement Pack, because the black pillar boxes are integrated right into the videos (why?)

We had a legitimate reason for doing this--because of how the current FMV display code works--but it's not worth getting into because we plan to overhaul it in the future. And for the FMV Enhancement Pack update, the videos will no longer have baked-in pillarboxing.

> p.s. I've noticed that main game logo is cropped or enlarged too much in 1280x1024 too...

This is correct behavior and will not be changed. The main menu and save screens use special code and will be cropped at aspect ratios < 16:9.

-------------------------------

Here is also a still image preview of the next FMV Enhancement Pack update:

### PlayStation 2
![image](https://user-images.githubusercontent.com/27395260/112239866-aa0f4600-8c1d-11eb-9555-29962d703282.png)

### FMV Enhancement Pack (current)
![image](https://user-images.githubusercontent.com/27395260/112239876-aed3fa00-8c1d-11eb-9939-e7016b8ea57e.png)

### FMV Enhancement Pack (next update)
![image](https://user-images.githubusercontent.com/27395260/112239886-b4c9db00-8c1d-11eb-9b6c-0d8268147943.png)

**Comment by Psycho-A** [comment]
> We have been discussing ways to completely revamp the FMV display feature for the project, so all of this shouldn't be a concern in the near future.
> 
> If all goes to plan, with this revamp, FMV display modes will behave similar to FullscreenImages and it will be consolidated into one feature (so instead of FMVWidescreenMode and FMVWidescreenEnhancementPackCompatibility, it will be just one feature like FMVDisplayMode).

That would be very nice :)
I even think that using some single param for images and videos will be more convenient for users because no reason to have different settings here as they both drawn at similar aspect ratio and user rather expects the same appearance behavior...

> This is correct behavior and will not be changed. The main menu and save screens use special code and will be cropped at aspect ratios < 16:9.

But can be this logo just a little reduced in size (without editing code) just to fit a bit better to such screens? I don't think this will be too noticeable in 16:9.

> FMV Enhancement Pack (next update)

Looks good!

**Comment by Polymega** [comment]
> But can be this logo just a little reduced in size (without editing code) just to fit a bit better to such screens? I don't think this will be too noticeable in 16:9.

It can't. If the logo didn't fade from black-to-white we maybe could, but we can't because of this. From what I remember with our tests, only in 5:4 does the logo become very close to the screen edge. You can play in another aspect ratio to prevent this (we recommend 4:3 or 16:9).

**Comment by Psycho-A** [comment]
> It can't. If the logo didn't fade from black-to-white we maybe could, but we can't because of this. From what I remember with our tests, only in 5:4 does the logo become very close to the screen edge. You can play in another aspect ratio to prevent this (we recommend 4:3 or 16:9).

How the fading effect is connected with logo size? The original logo was narrower, but it had that effect too... I don't play with that aspect ratio myself - I just worry about ones who do. The 5:4 monitors are **more** popular than 4:3 because lots of middle 00s-10s budget LCDs had 1280x1024 resolution.

**Comment by Polymega** [comment]
> How the fading effect is connected with logo size? The original logo was narrower, but it had that effect too...

You're right. My hands are involved with many aspects of this project and I forget details about things we've done a long time ago. I'm not going to adjust the logo size as it's not actually cropped in 5:4; it's just close to the edges. We recommend people play in 4:3 or 16:9 and [warn that there may be visual issues if played in any other aspect ratio.](http://enhanced.townofsilenthill.com/SH2/troubleshoot.htm#do-i-have-to-play-silent-hill-2-enhanced-edition-in-169-widescreen)

However, I have shifted the logo over slightly to the right to fix it being off-centered, which is most noticeable when played in 5:4.

-----------------------------------------

Default languages: [Default Languages.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6214780/Default.Languages.zip)

@belek666 Polish: [start01p.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6214781/start01p.zip)

@RobertoSAG Portuguese: [start01p.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6214784/start01p.zip)

@HellbringerOnline Russian: [start01r.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6214785/start01r.zip)

@Tanshaydar Turkish: [start01t.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6214786/start01t.zip)

**Comment by RobertoSAG** [comment]
> > How the fading effect is connected with logo size? The original logo was narrower, but it had that effect too...
> 
> You're right. My hands are involved with many aspects of this project and I forget details about things we've done a long time ago. I'm not going to adjust the logo size as it's not actually cropped in 5:4; it's just close to the edges. We recommend people play in 4:3 or 16:9 and [warn that there may be visual issues if played in any other aspect ratio.](http://enhanced.townofsilenthill.com/SH2/troubleshoot.htm#do-i-have-to-play-silent-hill-2-enhanced-edition-in-169-widescreen)
> 
> However, I have shifted the logo over slightly to the right to fix it being off-centered, which is most noticeable when played in 5:4.
> 
> Default languages: [Default Languages.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6214780/Default.Languages.zip)
> 
> @belek666 Polish: [start01p.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6214781/start01p.zip)
> 
> @RobertoSAG Portuguese: [start01p.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6214784/start01p.zip)
> 
> @HellbringerOnline Russian: [start01r.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6214785/start01r.zip)
> 
> @Tanshaydar Turkish: [start01t.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6214786/start01t.zip)

Are these new files for our translations or the same ones you provided before?

**Comment by Polymega** [comment]
They're new files. I fixed a small issue where the SH2 logo was slightly off-center. The translations haven't been touched; just the logo was slightly shifted.

**Comment by RobertoSAG** [comment]
> They're new files. I fixed a small issue where the SH2 logo was slightly off-center. The translations haven't been touched*, just the logo was slightly shifted.
> 
> *You'll still need to use both the new `start00` and `start01*` files together in the download package, though.

Thank you for them! I'll update the translation pack.

**Comment by HellbringerOnline** [comment]
@Polymega Can you remind me what the differens between start00 and start01?

**Comment by Polymega** [comment]
start00 is the background image for the main menu (the black SH2 logo and the "water" effect behind it).

start01 is the main menu choices and a white SH2 logo that fades in and out.

You need both to work together. For the package I linked you, you'd change the "r" in `start01r` to whatever language you are replacing for the Russian pack.

**Comment by HellbringerOnline** [comment]
@Polymega thank you, i updated the files

**Comment by HellbringerOnline** [comment]
> > > How the fading effect is connected with logo size? The original logo was narrower, but it had that effect too...
> > 
> > 
> > You're right. My hands are involved with many aspects of this project and I forget details about things we've done a long time ago. I'm not going to adjust the logo size as it's not actually cropped in 5:4; it's just close to the edges. We recommend people play in 4:3 or 16:9 and [warn that there may be visual issues if played in any other aspect ratio.](http://enhanced.townofsilenthill.com/SH2/troubleshoot.htm#do-i-have-to-play-silent-hill-2-enhanced-edition-in-169-widescreen)
> > However, I have shifted the logo over slightly to the right to fix it being off-centered, which is most noticeable when played in 5:4.
> > Default languages: [Default Languages.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6214780/Default.Languages.zip)
> > @belek666 Polish: [start01p.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6214781/start01p.zip)
> > @RobertoSAG Portuguese: [start01p.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6214784/start01p.zip)
> > @HellbringerOnline Russian: [start01r.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6214785/start01r.zip)
> > @Tanshaydar Turkish: [start01t.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/6214786/start01t.zip)
> 
> Are these new files for our translations or the same ones you provided before?

Hello, I got a message that the files `start00.tex` and `start01.tex` have stopped working. I haven't played the game for a long time, but downloaded the files, replaced them, but the translation hasn't changed. The folder `*\Silent Hill 2\sh2e\text` where the 
`exe_str.txt` text file was placed has also disappeared. How do I get everything to work again?

**Comment by Polymega** [comment]
> The folder *\Silent Hill 2\sh2e\text where the exe_str.txt text file was placed has also disappeared.

A lot has changed with the project since then. For starters, we no longer use `exe_str.txt` (so the `*\Silent Hill 2\sh2e\text` is also no longer used).

The contents found in the `exe_str.txt` file are now found in the `d3d8.res` file. Just note that, if someone updated the SH2 Enhancements module, this `d3d8.res` will be overwritten. So you will need to instruct people to hold onto their Russian translated `d3d8.res` file after updating the main module.

> Hello, I got a message that the files start00.tex and start01.tex have stopped working. I haven't played the game for a long time, but downloaded the files, replaced them, but the translation hasn't changed.

Just to clarify: The packages I linked a few posts above are using "dummy" file names. In the Russian package, I named the file that displays the text as `start01r` (the "r" stands for Russian). You will need to rename this file accordingly. 

For example: If you are replacing the English text in the game with Russian, you would rename it to `start01e` (and `start01` as the default language in the game is set to English). If you are replacing the Spanish text with Russian, you would rename is `start01s`. If you are replacing French you would rename it `start01f`.
